experiment/src/cortex/experiment/util.clj

```diff
@@ -6,6 +6,39 @@
   (:import [java.io File]))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Data encoding utils
+
+(defn label->one-hot
+  "Given a vector of class-names and a label, return a one-hot vector based on
+  the position in class-names.
+  E.g.  (label->vec [:a :b :c :d] :b) => [0 1 0 0]"
+  [class-names label]
+  (let [num-classes (count class-names)
+        src-vec (vec (repeat num-classes 0))
+        label-idx (.indexOf class-names label)]
+    (when (= -1 label-idx)
+      (throw (ex-info "Label not in classes for label->one-hot"
+                      {:class-names class-names :label label})))
+    (assoc src-vec label-idx 1)))
+
+
+(defn one-hot-encoding
+  "Given a dataset and a list of categorical features, returns a new dataset with these
+  features encoded into one-hot indicators
+  E.g. (one-hot-encoding [{:a :left} {:a :right} {:a :top}] [:a])
+         => [{:a_0 1 :a_1 0 :a_2 0} {:a_0 0 :a_1 1 :a_2 0} {:a_0 0 :a_1 0 :a_2 1}]"
+  [dataset features]
+  (reduce (fn [mapseq key]
+            (let [classes (set (map key mapseq))
+                  new-keys (for [i (range (count classes))]
+                             (keyword (str (name key) "_" i)))] ; a_0, a_1, a_2, etc.
+              (map (fn [elem] (->> (label->one-hot (vec classes) (key elem))
```

Comment:
In this case I think it's possible to only call `vec` once, rather than once per element.